### DISCUSSION TITLE
DBErrors.log

### DISCUSSION THREAD

**Discussion by zaicopx** [TrinityCore/TrinityCore#28319] [Q&A]
How to fix this:
Script '0' is referenced by the database, but does not exist in the core!
Script '0' is referenced by the database, but does not exist in the core!

What is Script '0' ??

**Comment by CraftedRO** [comment]
It means you inserted a script like this
you can read more here https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130125/spell_script_names
Example :
```sql
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
(50526, 'spell_zaicopx_startup_error');
```
but this spell doesn't exist in core \src\server\scripts\x.cpp files and can't be loaded so its ignored.

you can delete that spell from database like this and your error shoud be gone:
```sql
DELETE FROM `spell_script_names` WHERE `ScriptName` = 'spell_zaicopx_startup_error';
```
(replacing spell_zaicopx_startup_error with 0 if that's the name of your script)

or if you don't know the script name but you know spell_id for that script probably something like this:

```sql
DELETE FROM `spell_script_names` WHERE `spell_id` = 'XXXXX';
```
(replacing XXXXX with your spell id that trow a startup error)

As a bottom line I don't know also why some scripts are registered in spell_script_names and others are not but still present in core and doesn't trow any error.
